app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,40 @@
+require 'bcrypt'
+class PasswordResetsController < ApplicationController
+  def new
+    @page_title = "Reset Password | Haxxor News"
+  end
+  
+  def create
+    @user = User.find_by(email: params[:email])
+    if @user
+      flash[:notice] = "We've sent a reset password link to the given email."
+      PasswordResetMailer.with(user: @user, token: @user.password_token).reset_email.deliver_now
```

Comment:
I can't remember, but I think you'll generally use `deliver_later` so the controller can move on, and the mailer can work in the background